### PR TITLE
The cask code42-crashplan installs the Enterprise version of the soft…

### PR DESCRIPTION
**Pull request by Jarli01**
…ware - this installs the SMB version

Due to software differences the Enterprise version is not compatible with SMB users, thus a different cask is required. I'm not certain about the link location that is used in the existing cask and if one exists for the SMB client, so am providing this as a direct link.

After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.

Some particular error C: 5: 3: about anything I put in for the comment not matching the download URL, happy to fix this, just not sure what needs to be here. 

- [X] The commit message includes the cask’s name and version.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked the cask was not [already refused].
- [X] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/code42-crashplan-smb.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+cask 'code42-crashplan-smb' do
+  version '6.9.4'
+  sha256 '2f1d3131bdec06f879eb74456afcaaebc09d3a7a4901a8f7458189ff3efec5bb'
+
+  # download.code42.com was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/code42-crashplan-smb.rb]: This comment does not reflect the `url`.

File: Casks/code42-crashplan-smb.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  url "https://www.crashplanpro.com/client/installers/CrashPlanSmb_6.9.4_1525200006694_502_Mac.dmg"
```
Comment (vitorgalvao) [inline Casks/code42-crashplan-smb.rb]: Where did you get this from?

Comment (vitorgalvao) [reply Casks/code42-crashplan-smb.rb]: Interpolate the version.

File: Casks/code42-crashplan-smb.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  appcast 'https://support.code42.com/Release_Notes/CrashPlan_for_Small_Business_release_notes'
```
Comment (vitorgalvao) [inline Casks/code42-crashplan-smb.rb]: I don’t see this version on this page.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
> I'm not certain about the link location that is used in the existing cask and if one exists for the SMB client

So where are you getting it from?

**Comment by Jarli01** [comment]
Directly as show in the install url. What I meant about the link is I'm not certain if there is a version of the link that would work like with the other cask for the Enterprise version.

**Comment by Jarli01** [comment]
This is the direct download

https://www.crashplanpro.com/client/installers/CrashPlanSmb_6.9.4_1525200006694_502_Mac.dmg

File: Casks/code42-crashplan-smb.rb (hunk 2 above)
Comment (Jarli01) [reply Casks/code42-crashplan-smb.rb]: I'm not sure how or if there is a link that would work.

File: Casks/code42-crashplan-smb.rb (hunk 3 above)
Comment (Jarli01) [reply Casks/code42-crashplan-smb.rb]: It's crashplan for small business.

**Comment by vitorgalvao** [comment]
> Directly as show in the install url.

What install URL? I don’t know what you’re looking at or what you’re referencing.

> What I meant about the link is I'm not certain if there is a version of the link that would work like with the other cask for the Enterprise version.

What does that mean, “work like with the other cask”? What’s the difference?

> This is the direct download

So you’ve said, but I still have no clue where you’re getting it from.

Also, please comment directly on the other comments. This is dividing the conversation and making it harder.

**Comment by Jarli01** [comment]
> > I'm not certain about the link location that is used in the existing cask and if one exists for the SMB client
> 
> So where are you getting it from?



> > Directly as show in the install url.
> 
> What install URL? I don’t know what you’re looking at or what you’re referencing.

https://www.crashplanpro.com/client/installers/CrashPlanSmb_6.9.4_1525200006694_502_Mac.dmg

^ Install URL

> 
> > What I meant about the link is I'm not certain if there is a version of the link that would work like with the other cask for the Enterprise version.
> 
> What does that mean, “work like with the other cask”? What’s the difference?

This cask > https://github.com/Homebrew/homebrew-cask/blob/master/Casks/code42-crashplan.rb
 
> 
> > This is the direct download
> 
> So you’ve said, but I still have no clue where you’re getting it from.
> 
> Also, please comment directly on the other comments. This is dividing the conversation and making it harder.

**Comment by vitorgalvao** [comment]
> ^ Install URL

Please stop posting the same link over and over again. My question is **where did you get it FROM**. What’s the page you it **FROM**, and **WHERE** on the page is it located?

**Comment by Jarli01** [comment]
> > ^ Install URL
> 
> Please stop posting the same link over and over again. My question is **WHERE did you get it FROM**. What’s the page where that link is located, and where on said page?

You're not really thinking are you, that is the website where the file is downloaded from. 

If you'd like to see a generic version of the website: https://www.crashplan.com/en-us/download/

File: Casks/code42-crashplan-smb.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/code42-crashplan-smb.rb]: Again, this comment does not reflect the `url`. Have you read the contents of the comment? Do you see how the domain it refers to is not the same in the `url`?

**Comment by vitorgalvao** [comment]
> You're not really thinking are you

You’re not doing yourself any favours by spewing insults, particularly when you’re making multiple wrong decisions with the submission, such as marking as resolved a matter that was not, where you copy and pasted *blindly* a comment from another cask that does not apply here; haven’t followed the token reference; and didn’t even add a file extension.

And you clearly copy pasted the `appcast` as well, so I have to check if it’s correct, hence the question.

> that is the website where the file is downloaded from.
>
> If you'd like to see a generic version of the website: https://www.crashplan.com/en-us/download/

And nowhere in that page, looking at the source, is the URL linked directly, so I asked you the steps to get to it.

**Comment by core-code** [comment]
to make matters even more complicated, there supposedly is a third edition of "codeplan", besides Enterprise and SMB there is also the "Consumer" version though i am not sure if it still exists. i've tried quite hard to find and compare those different versions, but have given up. their website sucks just too much and it seems to be impossible to find definitive facts at least if you aren't a customer of them.

**Comment by core-code** [comment]
@Jarli01 are you planning on providing the requested information, so that this can be merged?

**Comment by core-code** [comment]
although this has been closed, i propose renaming the existing crashplan cask to make it obvious that it installs the enterprise version in contrast to the SMB version

**Comment by vitorgalvao** [comment]
> i propose renaming the existing crashplan cask to make it obvious that it installs the enterprise version

I don’t mind, though it may cause some disruption.

```
code42-crashplan (added 203 days ago)
30 days: 1,200 (#108)
90 days: 2,431 (#142)
365 days: 2,470 (#313)
```

**Comment by core-code** [comment]
ok, lets not cause any disruption then ;)

**Comment by vitorgalvao** [comment]
I’m ambivalent. On the one hand, the cask is under a year old so the time to cause the disruption is as soon as possible. On the other hand there doesn’t seem to be any problem now, and by the time there is we may have implemented the functionality to handle renames/migrations.

**Comment by core-code** [comment]
unless someone complains, probably better to wait until renames are handled better in homebrew

**Comment by vitorgalvao** [comment]
> until renames are handled better in homebrew

Just Homebrew Cask. Homebrew handles formula renames well, but the functionality needs to be ported to Cask.

**Comment by core-code** [comment]
ah ok, thanks, didn't know that

**Pull request closed without merging**